scripts/typings_tests/test_typings.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * The following tests are to ensure type declarations are maintained.
+ *
+ * Test:
+ *   - Files include reference paths to the Protractor's generated typings file.
+ *   - Each file is transpiled using:
+ *        tsc {fileName}
+ *   - Files that follow the type declarations should produce no errors.
+ *   - Files that do not follow the type declarations should produce errors.
+ *     For error files, it is assumed that the format matches the following:
+ *        {fileName}({lineNumber}
+ *     and should have an error for each line from starting line using the types
+ *     to the ending line of the file.
```

Comment:
One more thing before merge - can you add WHY we need these? e.g. why we don't just trust that typescript works because we have these globals and ugly selenium wrapping.